packages/core/src/utils/ethapi.ts

```diff
@@ -250,4 +264,11 @@ export class EthApi extends Events {
       this.emit("accountSet");
     }
   };
+
+  private setNetwork = (newNetwork: string): void => {
+    if (newNetwork !== this.currentNetwork) {
+      this.currentNetwork = newNetwork;
+      this.emit("networkSet");
```

Comment:
should this emit the network id? `this.emit("networkSet", {networkId: newNetwork})`

Reply:
It doesn't really matter, you have immediate access to it anyways. I left this out for now but if we want to add it later should be fine.